twisted/internet/asyncioreactor.py

```diff
@@ -171,18 +171,18 @@ def addWriter(self, writer):
                                               self._readOrWrite, writer,
                                               False)
             self._writers[writer] = fd
-        except IOError as e:
+        except PermissionError:
             self._unregisterFDInAsyncio(fd)
-            if e.errno == errno.EPERM:
-                # epoll(7) doesn't support certain file descriptors,
-                # e.g. filesystem files, so for those we just poll
-                # continuously:
-                self._continuousPolling.addWriter(writer)
-            else:
-                raise
-        except OSError as e:
+            # epoll(7) doesn't support certain file descriptors,
+            # e.g. filesystem files, so for those we just poll
+            # continuously:
+            self._continuousPolling.addWriter(writer)
+        except BrokenPipeError:
             # The kqueuereactor will raise this if there is a broken pipe
```

Comment:
no, i really wouldn't :(  after a few minutes of thinking i can't see a better way to do this.  maybe it'd be good to extend the comment to mention that it's ok to discard `BrokenPipeError` because all it means is that the pipe got closed, and that EPIPE is to writable pipes as an empty read is to readable pipes? or add another comment, in the `except:` block, about how if that code runs it's really an error and not some vagary of asyncio/python 3/posix?